File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/VB/Direct2D/Paint2D/DrawingShape.vb
' Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports Microsoft.WindowsAPICodePack.DirectX.Direct2D1

Namespace D2DPaint
	Friend MustInherit Class DrawingShape
		Protected _parent As Paint2DForm
		Protected _fill As Boolean

		Protected Friend MustOverride Sub Draw(ByVal renderTarget As RenderTarget)
		Protected Friend Overridable Sub EndDraw()
		End Sub

		Protected Friend MustOverride WriteOnly Property EndPoint() As Point2F

		Protected Sub New(ByVal parent As Paint2DForm)
			Me._parent = parent
		End Sub

		Protected Sub New(ByVal parent As Paint2DForm, ByVal fill As Boolean)
			Me.New(parent)
			Me._fill = fill
		End Sub
	End Class
End Namespace

File: main/contrib/Windows API Code Pack 1.1/source/Samples/DirectX/CS/Direct2D/Paint2D/DrawingShape.cs
// Copyright (c) Microsoft Corporation.  All rights reserved.

using Microsoft.WindowsAPICodePack.DirectX.Direct2D1;

namespace D2DPaint
{
    abstract class DrawingShape
    {
        protected Paint2DForm _parent;
        protected bool _fill;

        protected internal abstract void Draw(RenderTarget renderTarget);
        protected internal virtual void EndDraw()
        { }

        protected internal abstract Point2F EndPoint
        {
            set;
        }

        protected DrawingShape(Paint2DForm parent)
        {
            this._parent = parent;
        }

        protected DrawingShape(Paint2DForm parent, bool fill) : this(parent)
        {
            this._fill = fill;
        }
    }
}